src/LVMGenerator.h

```diff
@@ -52,25 +59,82 @@ class RelationEncoder {
     }
 
     /** Get total number of relations */
-    size_t getSize() const {
+    size_t getRelationNumber() const {
         return relationMap.size();
     }
 
-    std::unique_ptr<LVMRelation>& operator[](size_t idx) {
+    /** Return ptr to relation based on index position */
+    std::unique_ptr<LVMRelation>& getRelation(size_t idx) {
         return relationMap[idx];
     }
 
+    /** Encode and return index position based on the operation */
+    template <class RamNode>
+    size_t getIndexPos(RamNode* node) {
+        const MinIndexSelection& orderSet = this->isa->getIndexes(node->getRelation());
+        SearchSignature signature = this->isa->getSearchSignature(node);
+        // A zero signature is equivalent as a full order signature.
+        if (signature == 0) {
+            signature = (1 << node->getRelation().getArity()) - 1;
+        }
+        return orderSet.getLexOrderNum(signature);
+    };
+
+    /** Encode string */
+    size_t encodeString(const std::string& str) {
+        return symbolTable.lookup(str);
+    }
+
+    /** Decode string */
+    std::string decodeString(size_t id) const {
+        return symbolTable.resolve(id);
+    }
+
+    /** Encode and store IODirctives. Return its id. */
```

Comment:
Typo, and  a suggestion for clarity
```suggestion
    /** Encode and store IODirectives. Returns the generated id. */
```